pkg/reconciler/pubsub/reconciler.go

```diff
@@ -61,15 +61,23 @@ const (
 	OpsJobOngoing            OpsJobStatus = "JOB_ONGOING"
 )
 
-func (c *PubSubBase) EnsureSubscription(ctx context.Context, owner kmeta.OwnerRefable, project, topic, subscription string) (OpsJobStatus, error) {
+func (c *PubSubBase) EnsureSubscriptionExists(ctx context.Context, owner kmeta.OwnerRefable, project, topic, subscription string) (OpsJobStatus, error) {
+	return c.ensureSubscriptionJob(ctx, operations.ActionExists, owner, project, topic, subscription)
+}
+
+func (c *PubSubBase) EnsureSubscriptionCreated(ctx context.Context, owner kmeta.OwnerRefable, project, topic, subscription string) (OpsJobStatus, error) {
 	return c.ensureSubscriptionJob(ctx, operations.ActionCreate, owner, project, topic, subscription)
 }
 
 func (c *PubSubBase) EnsureSubscriptionDeleted(ctx context.Context, owner kmeta.OwnerRefable, project, topic, subscription string) (OpsJobStatus, error) {
 	return c.ensureSubscriptionJob(ctx, operations.ActionDelete, owner, project, topic, subscription)
 }
 
-func (c *PubSubBase) EnsureTopic(ctx context.Context, owner kmeta.OwnerRefable, project, topic string) (OpsJobStatus, error) {
+func (c *PubSubBase) EnsureTopicExists(ctx context.Context, owner kmeta.OwnerRefable, project, topic string) (OpsJobStatus, error) {
+	return c.ensureTopicJob(ctx, operations.ActionExists, owner, project, topic)
+}
+
+func (c *PubSubBase) EnsureTopicCreated(ctx context.Context, owner kmeta.OwnerRefable, project, topic string) (OpsJobStatus, error) {
```

Comment:
Where are there EnsureTopic* functions used, or is that in future PRs?

Reply:
Future PR, because I split a larger PR up into parts. This is one part that uses Topic, but the impl of topic is later.